Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-17.ca-central-1.compute.internal
Sha: d0ae11551e95b6957762ac3696a399961e651449
Number of Error(s): 7
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
/home/centos/smoketest/PR-13367/HPCC-Platform/roxie/ccd/ccdmain.cpp:667:13: error: \xe2\x80\x98regenerateConfig\xe2\x80\x99 was not declared in this scope
make[2]: *** [roxie/ccd/CMakeFiles/ccd.dir/build.make:167: roxie/ccd/CMakeFiles/ccd.dir/ccdmain.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:16015: roxie/ccd/CMakeFiles/ccd.dir/all] Error 2
make: *** [Makefile:163: all] Error 2

[Build error:](https://gist.github.com/5df10393b7a12d3345ba8bb471a55db1)
Build log [Build log](https://gist.github.com/HPCCSmoketest/5df10393b7a12d3345ba8bb471a55db1#file-result-20-02-24-10-12-22-log)

--- a/ecl/agentexec/agentexec.cpp
+++ b/ecl/agentexec/agentexec.cpp
@@ -260,7 +260,6 @@ int main(int argc, const char *argv[])
     Owned<IPropertyTree> config;
     try
     {
-        DBGLOG("AgentExec: Loading config file 'agentexec.xml'");
         config.setown(loadConfiguration(eclagentDefaultJson, argv, "EclAgent", "ECLAGENT", "agentexec.xml", nullptr));
     }
     catch (IException *e) 

--- a/roxie/ccd/ccdmain.cpp
+++ b/roxie/ccd/ccdmain.cpp
@@ -578,6 +578,7 @@ int STARTQUERY_API start_query(int argc, const char *argv[])
     }
     init_signals();
 
+    bool dumpArgs = false;
     for (unsigned i=0; i<(unsigned)argc; i++)
     {
         if (stricmp(argv[i], "--help")==0 ||
@@ -592,6 +593,8 @@ int STARTQUERY_API start_query(int argc, const char *argv[])
             argv[i] = "--csv";
         else if (strsame(argv[i], "-raw"))
             argv[i] = "--raw";
+        else if (strsame(argv[i], "-v"))
+            dumpArgs = true;
     }
 
     #ifdef _USE_CPPUNIT
@@ -654,6 +657,16 @@ int STARTQUERY_API start_query(int argc, const char *argv[])
         topologyFile.append(codeDirectory).append(PATHSEPCHAR).append("RoxieTopology.xml");
         useOldTopology = checkFileExists(topologyFile.str());
         topology = loadConfiguration(useOldTopology ? nullptr : defaultJson, argv, "Roxie", "ROXIE", topologyFile, nullptr);
+        if (dumpArgs)
+        {
+            for (unsigned i=0; i<(unsigned)argc; i++)
+            {
+                DBGLOG("Arg: %s", argv[i]);
+            }
+            StringBuffer jsonText;
+            regenerateConfig(jsonText, topology, "Roxie");
+            DBGLOG("Configuration: %s", jsonText.str());
+        }
         saveTopology();
         const char *channels = topology->queryProp("@channels");
         if (channels)

--- a/system/jlib/jptree.cpp
+++ b/system/jlib/jptree.cpp
@@ -7860,16 +7860,15 @@ static void applyCommandLineOption(IPropertyTree * config, const char * option)
     }
 }
 
-static void displayConfig(IPropertyTree * config, const char * componentTag)
+jlib_decl StringBuffer & regenerateConfig(StringBuffer &jsonText, IPropertyTree * config, const char * componentTag)
 {
     Owned<IPropertyTree> recreated = createPTree();
     Owned<IPropertyTree> json = mapXmlConfigToJson(config);
     recreated->setProp("version", currentVersion);
     recreated->addPropTree(componentTag, json.getClear());
 
-    StringBuffer jsonText;
     toJSON(recreated, jsonText, 0, JSON_SortTags|JSON_Format);
-    printf("%s\n", jsonText.str());
+    return jsonText;
 }
 
 jlib_decl IPropertyTree * loadConfiguration(const char * defaultYaml, const char * * argv, const char * componentTag, const char * envPrefix, const char * legacyFilename, IPropertyTree * (mapper)(IPropertyTree *))
@@ -7902,7 +7901,9 @@ jlib_decl IPropertyTree * loadConfiguration(const char * defaultYaml, const char
         }
         else if (strsame(cur, "--init"))
         {
-            displayConfig(config, componentTag);
+            StringBuffer jsonText;
+            regenerateConfig(jsonText, config, componentTag);
+            printf("%s\n", jsonText.str());
             exit(0);
         }
         else if (strsame(cur, "--outputconfig"))
@@ -7955,7 +7956,9 @@ jlib_decl IPropertyTree * loadConfiguration(const char * defaultYaml, const char
 
     if (outputConfig)
     {
-        displayConfig(config, componentTag);
+        StringBuffer jsonText;
+        regenerateConfig(jsonText, config, componentTag);
+        printf("%s\n", jsonText.str());
         exit(0);
     }
 

--- a/system/jlib/jptree.hpp
+++ b/system/jlib/jptree.hpp
@@ -289,5 +289,6 @@ inline static bool isValidXPathChr(char c)
 }
 
 jlib_decl IPropertyTree * loadConfiguration(const char * defaultYaml, const char * * argv, const char * componentTag, const char * envPrefix, const char * legacyFilename, IPropertyTree * (mapper)(IPropertyTree *));
+jlib_decl StringBuffer & regenerateConfig(StringBuffer &jsonText, IPropertyTree * config, const char * componentTag);
 
 #endif

